packages/near-fast-auth-signer/src/api/index.ts

```diff
@@ -2,11 +2,47 @@ import { captureException } from '@sentry/react';
 import { KeyPair } from 'near-api-js';
 
 import { withTimeout } from '../utils';
-import { network } from '../utils/config';
+import { network, networkId } from '../utils/config';
 import {
   CLAIM, getUserCredentialsFrpSignature
 } from '../utils/mpc-service';
 
+const fetchAccountIdFromQueryApi = async (publicKey: string): Promise<string | null> => {
+  const query = `query AccountIdByPublicKey {
+    dataplatform_near_access_keys_v1_access_keys_v1(
+      where: {public_key: {_eq: "${publicKey}"}, deleted_by_receipt_id: {_is_null: true}, account_deleted_by_receipt_id: {_is_null: true}}
+    ) {
+      account_id
+      public_key
+      created_by_receipt_id
+      last_updated_block_height
+    }
+  }`;
+
+  try {
+    const res = await fetch(network.fastAuth.queryApiUrl, {
+      method:  'POST',
+      headers: { 'x-hasura-role': 'dataplatform_near' },
+      body:    JSON.stringify({
+        query,
+        variables:     {},
+        operationName: 'AccountIdByPublicKey',
+      }),
+    });
+    if (res) {
```

Comment:
@hcho112 I suggest checking for `res.ok` here instead of `res`, as `res` will always be truthy even for error responses. Using `res.ok` ensures that we specifically verify the success of the HTTP request, based on its status code. This helps to handle both successful and error responses appropriately.